src/features/blacksmith/components/CraftingItems.tsx

```diff
@@ -14,23 +14,25 @@ import { Craftable } from "features/game/events/craft";
 
 interface Props {
   items: Partial<Record<InventoryItemName, Craftable>>;
+  isBulk: boolean;
 }
 
-export const CraftingItems: React.FC<Props> = ({ items }) => {
+export const CraftingItems: React.FC<Props> = ({ items, isBulk = false }) => {
   const [selected, setSelected] = useState<Craftable>(Object.values(items)[0]);
 
   const { state, dispatcher, shortcutItem } = useContext(Context);
   const inventory = state.inventory;
 
-  const hasIngredients = selected.ingredients.every(
-    (ingredient) => (inventory[ingredient.item] || 0) >= ingredient.amount
+  const lessIngredients = (amount = 1) => selected.ingredients.some(
+    (ingredient) => (inventory[ingredient.item] || 0) < ingredient.amount * amount,
   );
-  const hasFunds = state.balance >= selected.price;
+  const lessFunds = (amount = 1) => state.balance < selected.price * amount;
 
-  const craft = () => {
+  const craft = (e, amount = 1) => {
```

Comment:
Consider omitting the click event parameter (`e`) as it is not needed for this function.

Below we can call it as `onClick={() => craft(10)}`

Reply:
Thanks for the feedback! 

> See below, you can still click "Buy 1" and "Buy 10" on the potatoes even though I have no potatoes

I forgot to default to 0 if selected is `undefined`

> Below we can call it as onClick={() => craft(10)}

My bad. At first single craft had errors because it accepted `SyntheticBaseEvent` as argument. Then I saw I still had  `onClick=${craft}` 😅